tornado-drivers/spirv/src/main/java/uk/ac/manchester/tornado/drivers/spirv/SPIRVDeviceContext.java

```diff
@@ -80,6 +84,11 @@ protected SPIRVDeviceContext(SPIRVDevice device, SPIRVContext context) {
         init(device);
         this.spirvContext = context;
         this.executionIds = Collections.synchronizedSet(new HashSet<>());
+        if (isDeviceContextLevelZero()) {
+            this.powerMetric = new SPIRVLevelZeroPowerMetric(this);
```

Comment:
I suggest renaming this field with: 

```suggestion
            this.powerMetricHandler = new SPIRVLevelZeroPowerMetric(this);
```